transpire/internal/config.py

```diff
@@ -195,3 +195,22 @@ def from_cwd(cls, cwd: Path = Path.cwd()) -> "ClusterConfig":
                 "cluster.toml not found up to current git or fs boundary"
             )
         return cls.from_cwd(cwd.parent)
+
+def get_config(name: str, module_name: str | None = None, cwd: Path = Path.cwd()) -> ModuleType:
+    if module_name:
+        cluster_toml = cwd / "cluster.toml"
+        if cluster_toml.exists():
+            return load_py_module_from_file(module_name, cluster_toml, name)
+        if cwd.is_mount() or (cwd / ".git").exists():
+            raise FileNotFoundError(
+                "cluster.toml not found up to current git or fs boundary"
+            )
```

Comment:
When this if statement branch falls through (we're not at the root already, and cluster_toml doesn't exist) there is no return statement or recursion, so this function will return `None`. This shouldn't pass typecheck I think!

Reply:
Yes I was sloppy! I've refactored the two cases.